### PR TITLE
Fix ARM build in general and particularly MacOS cases.

### PR DESCRIPTION
**Pull request by devnexen**
Forcing keeping arch/target value as it s lost most of the time (empty
values).
Forcing no x86 build if this is arm thus not building mistakenly x86
instrinsics.

### CODE REVIEW

**Comment by devnexen** [comment]
Note better to build with MACOS_USE_DEPENDENCIES as off since these are x86 binaries.

File: CMakeLists.txt [hunk 1]
```diff
@@ -299,7 +306,9 @@ else ()
     ADD_CHECK_CXX_COMPILER_FLAG(CMAKE_CXX_FLAGS CXX_WARN_LOGICAL_OP -Wlogical-op)
     ADD_CHECK_CXX_COMPILER_FLAG(CMAKE_CXX_FLAGS CXX_WARN_RANGE_LOOP_ANALYSIS -Wrange-loop-analysis)
     ADD_CHECK_CXX_COMPILER_FLAG(CMAKE_CXX_FLAGS CXX_WARN_TAUTOLOGICAL_ZERO_COMPARE -Wtautological-unsigned-zero-compare)
-    ADD_CHECK_CXX_COMPILER_FLAG(CMAKE_CXX_FLAGS CXX_WARN_OLD_STYLE_CAST -Wold-style-cast)
+    if (NOT APPLE)
+        ADD_CHECK_CXX_COMPILER_FLAG(CMAKE_CXX_FLAGS CXX_WARN_OLD_STYLE_CAST -Wold-style-cast)
```
Comment (duncanspumpkin) [inline CMakeLists.txt:310]: Why this change? We shouldn't need to use old style casts on any platform

Comment (devnexen) [reply CMakeLists.txt:310]: I got this because of openssl and its old C cast style from its headers :-)

Comment (tupaschoal) [reply CMakeLists.txt:313]: Then we need to make sure SSL is identified as a system dependency and not the actual code

Comment (janisozaur) [reply CMakeLists.txt:313]: @devnexen can you address that, please?

Comment (tupaschoal) [reply CMakeLists.txt:313]: Thing is that we don't want to disable this warning. We need to find out where in the MacOS build the openSSL and other library are being included and tell the build system that it is a system include

File: src/openrct2/CMakeLists.txt [hunk 2]
```diff
@@ -169,6 +170,12 @@ if (UNIX AND NOT ${CMAKE_SYSTEM_NAME} MATCHES "BSD")
 endif ()
 
 set(THREADS_PREFER_PTHREAD_FLAG ON)
+if (APPLE)
+	set(CMAKE_THREAD_LIBS_INIT "-lpthread")
+	set(CMAKE_HAVE_THREADS_LIBRARY 1)
+	set(CMAKE_USE_WIN32_THREADS_INIT 1)
```
Comment (adam-bloom) [inline src/openrct2/CMakeLists.txt:176]: why is a WIN32 cmake variable being set for macOS builds?

File: CMakeLists.txt [hunk 3]
```diff
@@ -20,8 +20,15 @@ if (CCache_FOUND)
 endif (CCache_FOUND)
 
 if (APPLE)
-    set(CMAKE_OSX_ARCHITECTURES "x86_64")
-    set(CMAKE_OSX_DEPLOYMENT_TARGET "10.14")
+    execute_process(COMMAND /usr/bin/uname -m OUTPUT_VARIABLE UNAME)
+    # oddily, cmake_system_processor can be empty
```
Comment (tupaschoal) [inline CMakeLists.txt:24]:
```suggestion
    # oddly, cmake_system_processor can be empty
```

**Comment by Gymnasiast** [comment]
@adam-bloom Has everything you wanted to port now been ported?

**Comment by adam-bloom** [comment]
> @adam-bloom Has everything you wanted to port now been ported?

Yes, I think we're good now.
